palantir-java-format-native/build.gradle

```diff
@@ -0,0 +1,111 @@
+
+buildscript {
+    repositories {
+        mavenCentral() { metadataSources { mavenPom(); ignoreGradleMetadataRedirection() } }
+        gradlePluginPortal() { metadataSources { mavenPom(); ignoreGradleMetadataRedirection() } }
+    }
+    dependencies {
+        classpath 'org.graalvm.buildtools:native-gradle-plugin:0.10.4'
+        classpath 'com.palantir.gradle.utils:platform:0.10.0'
+    }
+}
+
+
+apply plugin: 'java-library'
+apply plugin: 'org.graalvm.buildtools.native'
+apply plugin: 'com.palantir.external-publish-custom'
+
+dependencies {
+    implementation project(':palantir-java-format')
+}
+
+import com.palantir.platform.OperatingSystem
+import com.palantir.platform.Architecture
+
+def operatingSystem = OperatingSystem.get()
+def architecture = Architecture.get()
+
+def exports = [
+        'jdk.compiler/com.sun.tools.javac.file',
+        'jdk.compiler/com.sun.tools.javac.main',
+        'jdk.compiler/com.sun.tools.javac.parser',
+        'jdk.compiler/com.sun.tools.javac.tree',
+        'jdk.compiler/com.sun.tools.javac.util',
+        'jdk.compiler/com.sun.tools.javac.code',
+        'jdk.compiler/com.sun.tools.javac.api'
+]
+
+def jvmArgList = exports.collect { value -> "--add-exports=${value}=ALL-UNNAMED".toString() }
+
+
+graalvmNative {
+    binaries {
+        main {
+            javaLauncher = javaToolchains.launcherFor {
+                languageVersion = JavaLanguageVersion.of(23)
+                vendor = JvmVendorSpec.GRAAL_VM
+            }
+            imageName = String.format('palantir-java-format-%s-%s', operatingSystem.uiName(), architecture.uiName())
+            mainClass = 'com.palantir.javaformat.java.Main'
+            fallback = false
+            sharedLibrary = false
+            classpath.setFrom(configurations.named('runtimeClasspath'))
+            configurationFileDirectories.from(file('src/main/resources/META-INF/native-image'))
+            buildArgs.addAll("-H:ConfigurationFileDirectories=${projectDir}/src/main/resources/META-INF/native-image",
+                    '-H:+UnlockExperimentalVMOptions',
+                    '-H:IncludeResourceBundles=com.sun.tools.javac.resources.compiler',
+                    '-H:IncludeResourceBundles=com.sun.tools.javac.resources.javac',
+                    '--no-fallback',
+                    '--initialize-at-build-time=com.sun.tools.javac.file.Locations',
+                    '-H:+ReportExceptionStackTraces',
+                    '-H:-UseContainerSupport',
+                    "-march=compatibility")
+            jvmArgs.addAll(jvmArgList)
+        }
+    }
+}
+
+
+// The image requires parameters to be passed to; eg. gw nativeRun -Pargs=--palantir,-i,<file>
+tasks.named("nativeRun", org.graalvm.buildtools.gradle.tasks.NativeRunTask) {
+    runtimeArgs.addAll(project.hasProperty('appArgs') ? project.appArgs.split(',') : [])
+}
+
+def osExtension = operatingSystem == OperatingSystem.MACOS ? 'bin' : 'exe'
+
+externalPublishing {
+    publishing {
```

Comment:
I don't think this is calling anything under `externalPublishing`. There is a single method [`publication`](https://github.com/palantir/gradle-external-publish-plugin/blob/8fe67d8680eb57e37442b417a8177a2a9e1658f2/src/main/java/com/palantir/gradle/externalpublish/ExternalPublishCustomExtension.java#L32) that needs to be called with a name and a closure to configure the maven publication. This is currently calling the `publications` extension that lives on `project` from the `maven-publish` plugin and the external publish plugin does not know to actually publish this one.

Reply:
I changed this in: https://github.com/palantir/palantir-java-format/pull/1208/commits/8456625f22b51b7930eb83adf6fb2ea640f5a5ab#diff-012bca426d515b4e67fdc32e660cf6ade9abd1d720f3f923e6b16101a72dc3c3R79